packages/react-avatar/src/components/Avatar/useAvatarStyles.ts

```diff
@@ -249,150 +207,281 @@ const useStyles = makeStyles({
-    state.size >= 120 && styles.badgeSizeGreaterEqualThan120,
-    state.size >= 128 && styles.badgeSizeGreaterEqualThan128,
-    state.badge.className,
-  );
-  state.image.className = mergeClasses(styles.image, state.image.className);
-  state.label.className = mergeClasses(
-    styles.label,
-    state.color === 'brand' && styles.labelBrand,
-    state.color === 'darkRed' && styles.labelDarkRed,
-    state.color === 'cranberry' && styles.labelCranberry,
-    state.color === 'red' && styles.labelRed,
-    state.color === 'pumpkin' && styles.labelPumpkin,
-    state.color === 'peach' && styles.labelPeach,
-    state.color === 'marigold' && styles.labelMarigold,
-    state.color === 'gold' && styles.labelGold,
-    state.color === 'brass' && styles.labelBrass,
-    state.color === 'brown' && styles.labelBrown,
-    state.color === 'forest' && styles.labelForest,
-    state.color === 'seafoam' && styles.labelSeafoam,
-    state.color === 'darkGreen' && styles.labelDarkGreen,
-    state.color === 'lightTeal' && styles.labelLightTeal,
-    state.color === 'teal' && styles.labelTeal,
-    state.color === 'steel' && styles.labelSteel,
-    state.color === 'blue' && styles.labelBlue,
-    state.color === 'royalBlue' && styles.labelRoyalBlue,
-    state.color === 'cornflower' && styles.labelCornflower,
-    state.color === 'navy' && styles.labelNavy,
-    state.color === 'lavender' && styles.labelLavender,
-    state.color === 'purple' && styles.labelPurple,
-    state.color === 'grape' && styles.labelGrape,
-    state.color === 'lilac' && styles.labelLilac,
-    state.color === 'pink' && styles.labelPink,
-    state.color === 'magenta' && styles.labelMagenta,
-    state.color === 'plum' && styles.labelPlum,
-    state.color === 'beige' && styles.labelBeige,
-    state.color === 'mink' && styles.labelMink,
-    state.color === 'platinum' && styles.labelPlatinum,
-    state.color === 'anchor' && styles.labelAnchor,
-    state.label.className,
-  );
+
+  if (state.badge) {
+    let badgeSizeClass: string;
+    switch (state.badge.size!) {
+      case 'smallest':
+      case 'smaller':
+        badgeSizeClass = styles.badgeSmaller;
+        break;
+      case 'small':
+      case 'medium':
+        badgeSizeClass = styles.badgeMedium;
+        break;
+      case 'large':
+        badgeSizeClass = styles.badgeLarge;
+        break;
+      case 'larger':
+      case 'largest':
+        badgeSizeClass = styles.badgeLarger;
+        break;
+    }
+
+    state.badge.className = mergeClasses(styles.badge, badgeSizeClass, state.badge.className);
+  }
+
+  if (state.image) {
+    state.image.className = mergeClasses(styles.image, state.image.className);
+  }
+
+  if (color === 'colorful') {
+    throw new Error('Expected useAvatar to replace color="colorful" with a named color');
+  }
+
+  const colorClasses = useColorStyles();
+  state.label.className = mergeClasses(styles.label, colorClasses[color], state.label.className);
```

Comment:
Should this be:

```suggestion
  if (state.label) {
    const colorClasses = useColorStyles();
    state.label.className = mergeClasses(styles.label, colorClasses[color], state.label.className);
  }
```

Reply:
state.label is never undefined. A default value is provided in useAvatar, and the AvatarState type doesn't allow it to be undefined (AvatarDefaultedProps includes 'label')